### PR TITLE
chore: implement experiment details header section using InfoBox [WEB-1064]

### PR DESCRIPTION
**Pull request by EmilyBonar**
## Description
Before:
Screenshot 2023-03-30 at 14-30-43 Experiment 2128 - Determined

After:
Screenshot 2023-03-30 at 14-23-24 Experiment 2128 - Determined

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: webui/react/src/pages/ExperimentDetails/ExperimentDetailsHeader.tsx [hunk 1]
```diff
@@ -418,72 +419,79 @@ const ExperimentDetailsHeader: React.FC<Props> = ({
     [experiment.state],
   );
 
+  const foldableRows: InfoRow[] = useMemo(() => {
+    const rows = [
+      {
+        content: experiment.description || <Typography.Text disabled>N/A</Typography.Text>,
```
Comment (keita-determined) [inline webui/react/src/pages/ExperimentDetails/ExperimentDetailsHeader.tsx:425]: layout gets broken with long texts

image

**Review by keita-determined — COMMENTED** [review]
not sure if this is introduced in this PR, but i get dup key warnings

image

File: webui/react/src/pages/ExperimentDetails/ExperimentDetailsHeader.tsx (hunk 1 above)
Comment (EmilyBonar) [reply webui/react/src/pages/ExperimentDetails/ExperimentDetailsHeader.tsx]: This issue should be fixed. The duplicate key warnings look to be coming from the Tags component, which this PR didn't touch, so I think that's out of scope.

**Review by keita-determined — APPROVED** [review]
LGMT

**Pull request merged**